invalidtxnrefs.go

```diff
@@ -0,0 +1,99 @@
+// Copyright 2021 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package main
+
+import (
+	"github.com/juju/errors"
+	"github.com/juju/mgo/v2"
+	"github.com/juju/mgo/v2/bson"
+)
+// InvalidTxnReferenceCleaner looks for transactions that reference documents that cannot be found,
+// as well as documents that do not have the transaction in their queue.
+// These cannot be resumed cleanly, so we just remove them.
+type InvalidTxnReferenceCleaner struct {
+	txns *mgo.Collection
+	db *mgo.Database
+
+	// acceptablyMissingCollections lists the collection names that we 'accept' if they are
+	// missing. Normally the only collection that things should 'disappear' is "metrics".
+	// Because they are deleted by a batch process. However,
+	//we make this field serviceable by allowing additional collections to be supplied.
```

Comment:
fix spacing
also, it seems the only place we initialise this, it is with an empty slice
and I can't see it used anywhere?
So we should just delete this for now?

Reply:
I'm happy to remove it. The question is whether we want to implement the original logic.
So the original idea was to check if the document is missing, and not cleanup the txn unless it is in this set. (or if you manually supply a collection that includes this).
eg:
```
$ mgopurge
Found transaction with missing document: txn-id references <collection> <doc> but the doc is missing
$ mgopurge --delete-missing collection
[success]
```
Metrics is the only collection we *know* has things go missing. Is it worth being safer here? We aren't deleting the transaction (just rolling it back to preparing), but we are effectively removing it.

I don't know if this is a setting nobody wants to have to deal with "just make it work", vs giving us feedback about something unexpected. Thoughts?